### PR TITLE
[Refactor] deprecate enable_pipeline_load

### PR DESCRIPTION
**Pull request by silverbullet233**
a continuous work after #30106

in this PR, I deprecate `enable_pipeline_load` configuration in FE and make all load use pipeline engine by default.

NOTE: **after this change, enable_pipeline_load will no longer take effect.**

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [x] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [x] Yes, this PR will result in a change in behavior.
- [ ] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [x] Policy changes: use new policy to replace old one, functionality automatically enabled
- [x] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 17 / 31 (54.84%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadLoadingTask.java|0|5|00.00%|[135, 138, 139, 165, 276]|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadingTaskPlanner.java|2|9|22.22%|[265, 266, 267, 268, 347, 348, 349]|
|:large_blue_circle:|com/starrocks/planner/StreamLoadScanNode.java|1|2|50.00%|[437]|
|:large_blue_circle:|com/starrocks/sql/LoadPlanner.java|8|9|88.89%|[222]|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/FileScanNode.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/InsertPlanner.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/DataSink.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by trueeyu — APPROVED** [review]

**Review by satanson — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/common/Config.java [hunk 1]
```diff
@@ -2026,6 +2026,7 @@ public class Config extends ConfigBase {
     /**
      * Enable pipeline engine load
      */
+    @Deprecated
     @ConfField(mutable = true)
     public static boolean enable_pipeline_load = true;
```
Comment (chaoyli) [inline fe/fe-core/src/main/java/com/starrocks/common/Config.java:2031]: Could you make it not visible?

**Review by chaoyli — APPROVED** [review]

**Pull request merged**